oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnectionRule.java

```diff
@@ -126,61 +124,56 @@ private void downloadSimilaritySearchPluginIfNotExists(String localPluginPath, S
-                String query = uri.getQuery();
-
-                String api_key = null;
-                String api_secret = null;
-                if (query != null) {
-                    api_key = query.split(",")[0].split("=")[1];
-                    api_secret = query.split(",")[1].split("=")[1];
-                }
-                elasticConnection = ElasticConnection.newBuilder()
-                        .withIndexPrefix(INDEX_PREFIX + System.currentTimeMillis())
-                        .withConnectionParameters(scheme, host, port)
-                        .withApiKeys(api_key, api_secret)
-                        .build();
-            } catch (URISyntaxException e) {
-                return null;
+        try {
+            URI uri = new URI(elasticConnectionString);
+            String host = uri.getHost();
+            String scheme = uri.getScheme();
+            int port = uri.getPort();
+            String query = uri.getQuery();
+
+            String api_key = null;
+            String api_secret = null;
+            if (query != null) {
+                api_key = query.split(",")[0].split("=")[1];
+                api_secret = query.split(",")[1].split("=")[1];
             }
+            return ElasticConnection.newBuilder()
+                    .withIndexPrefix(INDEX_PREFIX + System.currentTimeMillis())
+                    .withConnectionParameters(scheme, host, port)
+                    .withApiKeys(api_key, api_secret)
+                    .build();
+        } catch (URISyntaxException e) {
+            return null;
         }
-        return elasticConnection;
     }
 
     public ElasticConnection getElasticConnectionForDocker() {
-        if (elasticConnection == null) {
-            elasticConnection = ElasticConnection.newBuilder()
-                    .withIndexPrefix(INDEX_PREFIX + System.currentTimeMillis())
-                    .withConnectionParameters(ElasticConnection.DEFAULT_SCHEME,
-                            elastic.getContainerIpAddress(),
-                            elastic.getMappedPort(ElasticConnection.DEFAULT_PORT))
-                    .withApiKeys(null, null)
-                    .build();
-        }
-        return elasticConnection;
+        return getElasticConnectionForDocker(elastic.getContainerIpAddress(),
+                elastic.getMappedPort(ElasticConnection.DEFAULT_PORT));
+    }
+
+    public ElasticConnection getElasticConnectionForDocker(String containerIpAddress, int port) {
+        return ElasticConnection.newBuilder()
+                .withIndexPrefix(INDEX_PREFIX + System.currentTimeMillis())
+                .withConnectionParameters(ElasticConnection.DEFAULT_SCHEME,
+                        containerIpAddress, port)
+                .withApiKeys(null, null)
+                .build();
     }
 
-    public void closeElasticConnection() throws IOException {
+    public void closeElasticConnection(ElasticConnection elasticConnection) throws IOException {
```

Comment:
I couldn't find where this is getting called from now ?

Reply:
good catch! This is not needed anymore since the container gets removed when the tests are complete.

Reply:
@fabriziofortino 
I think not having a cleanup method that closes ElasticConnection and deletes test indexes might create problems if tests are run on a non docker bases elastic server - I am not sure if we would really need to do that. 
But maybe we can add TODO statement there in case it's needed in future.

Rest all looks good to me.

Reply:
You are right. I forgot it was possible to run the tests without testcontainers. I added back the delete index but I have moved it in `ElasticAbstractQueryTest`. I think it makes more sense to have it there.